File: Config/Generator/templates/1/scripts/BaseConfig.cs
using System;
using System.Collections.Generic;

namespace GameConfig
{
    public class BaseConfig<T, W>
    {
        public string ConfigName { get; private set; }

        public IReadOnlyDictionary<T, W> Data { private set; get; }

        public BaseConfig(string configName, IReadOnlyDictionary<T, W> data)
        {
            ConfigName = configName;
            Data = data;
        }

        public W Get(T key)
        {
            Data.TryGetValue(key, out var result);
            return result;
        }

        public W Get(T key, bool ifNullThrowException)
        {
            if (ifNullThrowException)
            {
                if (!Data.ContainsKey(key))
                    throw new Exception($"{ConfigName} not found => {key}");
            }

            Data.TryGetValue(key, out var result);
            return result;
        }
    }
}

File: Config/Generator/templates/2/scripts/BaseConfig.ts
export class BaseConfig<T, W> {
    private _configName: string;
    public get configName() {
        return this._configName;
    }

    private _data: Map<T, W>;
    public get data(): Map<T, W> {
        return this._data;
    };

    constructor(configName: string, sourceData: Map<T, W>) {
        this._configName = configName;
        this._data = sourceData;
    }

    public get(key: T, ifNullThrowError: boolean = false): W {
        if (ifNullThrowError) {
            if (!this._data.has(key))
                console.error(`${this._configName} not found => ${key}`);
        }
        return this._data.get(key);
    }
}